stravalib/model.py

```diff
@@ -432,379 +553,165 @@ class ActivityKudos(Athlete):
+    }
 
-    name = Attribute(str, (SUMMARY, DETAILED))  #: Name of lap
-    activity = EntityAttribute(
-        "Activity", (SUMMARY, DETAILED)
-    )  #: The associated :class:`stravalib.model.Activity`
-    athlete = EntityAttribute(
-        Athlete, (SUMMARY, DETAILED)
-    )  #: The associated :class:`stravalib.model.Athlete`
+    _unit_registry = {
+        'distance': 'meters',
+        'total_elevation_gain': 'meters',
+        'average_speed': 'm/s',
+        'max_speed': 'm/s'
+    }
 
-    elapsed_time = TimeIntervalAttribute(
-        (SUMMARY, DETAILED)
-    )  #: :class:`datetime.timedelta` of elapsed time for lap
-    moving_time = TimeIntervalAttribute(
-        (SUMMARY, DETAILED)
-    )  #: :class:`datetime.timedelta` of moving time for lap
-    start_date = TimestampAttribute(
-        (SUMMARY, DETAILED)
-    )  #: :class:`datetime.datetime` when lap was started in GMT
-    start_date_local = TimestampAttribute(
-        (SUMMARY, DETAILED), tzinfo=None
-    )  #: :class:`datetime.datetime` when lap was started local
-    distance = Attribute(
-        float, (SUMMARY, DETAILED), units=uh.meters
-    )  #: The distance for this lap.
-    start_index = Attribute(int, (SUMMARY, DETAILED))  #:
-    end_index = Attribute(int, (SUMMARY, DETAILED))  #:
-    total_elevation_gain = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-        units=uh.meters,
-    )  #: What is total elevation gain for lap
-    average_speed = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-        units=uh.meters_per_second,
-    )  #: Average speed for lap
-    max_speed = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-        units=uh.meters_per_second,
-    )  #: Max speed for lap
-    average_cadence = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Average cadence for lap
-    average_watts = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Average watts for lap
-    average_heartrate = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Average heartrate for lap
-    max_heartrate = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Max heartrate for lap
-    lap_index = Attribute(int, (SUMMARY, DETAILED))  #: Index of lap
-    device_watts = Attribute(
-        bool, (SUMMARY, DETAILED)
-    )  # true if the watts are from a power meter, false if estimated
-    pace_zone = Attribute(int, (SUMMARY, DETAILED))  #: (undocumented)
-    split = Attribute(int, (SUMMARY, DETAILED))  #: Split number
+    _naive_local = validator('start_date_local', allow_reuse=True)(naive_datetime)
 
+    _activity_extension = extend_types('activity', model_class='Activity')
```

Comment:
model_class sometimes needs to be given as str because the type is not defined (yet)